src/components/Link/Link.tsx

```diff
@@ -33,6 +34,13 @@ export function isCustomProps<T>(
 ): props is CustomLinkProps<T> {
   return 'asCustom' in props
 }
+// keyboard handler for 'link as a button'
+const handleKeyDown = (e: React.KeyboardEvent) => {
```

Comment:
I was hoping typing this as `const handleKeyDown: React.KeyboardEventHandler<HTMLAnchorElement>` would maybe avoid the need to cast e.target, but alas, it does not. So IMO, this is fine.